src/group_theory/nilpotent.lean

```diff
@@ -422,6 +425,18 @@ begin
   exact eq_bot_iff.mpr (λ x hx, subtype.ext (this x hx)),
 end
 
+/-- A the nilpotency class of a subgroup is less or equal the the nilpotency class of the group -/
+lemma subgroup.nilpotency_class_le (H : subgroup G) [hG : is_nilpotent G] :
+  group.nilpotency_class H ≤ group.nilpotency_class G :=
+begin
+  repeat { rw ← lower_central_series_length_eq_nilpotency_class },
+  apply nat.find_mono,
+  intros n hG,
+  have := lower_central_series_map_subtype_le H n,
+  simp only [hG, set_like.le_def, mem_map, forall_apply_eq_imp_iff₂, exists_imp_distrib] at this,
```

Comment:
Non-terminal `simp`. Will investigate, but suggestions are welcome.

Reply:
A non-terminal `simp only` is fine! So I don't think there is too much to investigate here, unless I misunderstood your intention